spikeinterface/comparison/addtemplatesrecording.py

```diff
@@ -0,0 +1,130 @@
+
+        if nbefore is None:
+            nbefore = np.argmax(np.max(np.abs(templates), axis=2), axis=1)
+        elif isnumeric(nbefore):
+            nbefore = [nbefore]*n_units 
+        else:
+            assert len(nbefore) == n_units
+
+        if amplitude_factor is None:
+            amplitude_factor = [[1.0]*len(sorting.get_unit_spike_train(unit_id)) for unit_id in sorting.unit_ids]
+
+        if target_recording is not None:
+            assert target_recording.get_num_segments() == sorting.get_num_segments()
+            target_recording.copy_metadata(self)
+
+        if t_max is None:
+            if target_recording is None:
+                t_max = [self.spike_vector['sample_ind'][-1] + templates.shape[1]]
+            else:
+                t_max = [target_recording.get_num_frames(segment_index) for segment_index in range(sorting.get_num_segments())]
+
+
+        for segment_index in range(sorting.get_num_segments()):
+            spikes = self.spike_vector[self.spike_vector['segment_ind'] == segment_index]
+            target_recording_segment = None if target_recording is None else target_recording._recording_segments[segment_index]
+            recording_segment = AddTemplatesRecordingSegment(self.sampling_frequency, 0, self.dtype, spikes, templates, nbefore,
+                                                             amplitude_factor, target_recording_segment, t_max[segment_index]) # TODO: t_start should not necessarily be 0.
+            self.add_recording_segment(recording_segment)
+
+
+
+class AddTemplatesRecordingSegment(BaseRecordingSegment):
+    """
+    TODO
+    """
+
+    def __init__(self, sampling_frequency: float, t_start: int, dtype, spike_vector: np.ndarray, templates: np.ndarray, nbefore: List[int],
+                 amplitude_factor: List[List[float]], target_recording: Union[BaseRecordingSegment, None] = None, t_max: Union[int, None] = None) -> None:
+
+        BaseRecordingSegment.__init__(self, sampling_frequency, t_start=t_start)
+        assert not (target_recording is None and t_max is None)
+
+        self.dtype = dtype
+        self.spike_vector = spike_vector
+        self.templates = templates
+        self.nbefore = nbefore
+        self.amplitude_factor = amplitude_factor
+        self.parent_recording = target_recording
+        self.t_max = target_recording.get_num_frames() if t_max is None else t_max
+
+    def get_traces(self, start_frame: Union[int, None] = None, end_frame: Union[int, None] = None,
+                   channel_indices: Union[List, None] = None) -> np.ndarray:
+        n_channels = self.templates.shape[2]
+        start_frame = 0 if start_frame is None else start_frame
+        end_frame = self.t_max if end_frame is None else end_frame
+
+        if self.parent_recording is not None:
+            traces = self.parent_recording.get_traces(start_frame, end_frame)
+        else:
+            traces = np.zeros([self.t_max, n_channels], dtype=self.dtype)
```

Comment:
Here you have a mistake it should be (end_frame, n_channels).

Reply:
But `t_max` is `end_frame` right?
If you are talking about the fact that this is a list rather than a tuple, it also works (I use list because I like how it looks).

I don't understand what the mistake is.

Reply:
Oups sorry I meant end_fram-start_frame
in that case when sdhould traces = np.zeros([endèframe-start_frame, n_channels], dtype=self.dtype)

Reply:
Ah you're right!
Changing it.